integration-tests/leveldb/src/main/java/org/apache/camel/quarkus/component/leveldb/it/LeveldbRouteBuilder.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.camel.quarkus.component.leveldb.it;
+
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import org.apache.camel.AggregationStrategy;
+import org.apache.camel.Exchange;
+import org.apache.camel.builder.RouteBuilder;
+import org.apache.camel.component.leveldb.LevelDBAggregationRepository;
+
+public class LeveldbRouteBuilder extends RouteBuilder {
+    public static final String DIRECT_START = "direct:start";
+    public static final String DIRECT_START_WITH_FAILURE = "direct:startWithFailure";
+    public static final String DIRECT_START_DEAD_LETTER = "direct:startDeadLetter";
+    public static final String MOCK_AGGREGATED = "mock:aggregated";
+    public static final String MOCK_RESULT = "mock:result";
+    public static final String MOCK_DEAD = "mock:dead";
+    public static final String DATA_FOLDER = "target/data";
+
+    private static AtomicInteger counter = new AtomicInteger(0);
+
+    @Override
+    public void configure() throws Exception {
+        LevelDBAggregationRepository repo = new LevelDBAggregationRepository("repo", DATA_FOLDER + "leveldb.dat");
+
+        from(DIRECT_START)
+                .aggregate(header("id"), new MyAggregationStrategy())
+                .completionSize(7).aggregationRepository(repo)
+                .to(MOCK_RESULT);
+
+        LevelDBAggregationRepository repoWithFailure = new LevelDBAggregationRepository("repoWithFailure",
+                DATA_FOLDER + "leveldbWithFailure.dat");
+
+        repoWithFailure.setUseRecovery(true);
+        repoWithFailure.setRecoveryInterval(500, TimeUnit.MILLISECONDS);
+
+        from(DIRECT_START_WITH_FAILURE)
+                .aggregate(header("id"), new MyAggregationStrategy())
+                .completionSize(7).aggregationRepository(repoWithFailure)
+                .to(MOCK_AGGREGATED)
+                .delay(1000)
```

Comment:
Why do we need the delay here?

Reply:
This configuration was copy/pasted from junit from camel-leveldb. I remeber that not having it there could change to order of messages, I'll test it once again.

Reply:
If it is a matter of ordering and if different orderings are correct outcomes of the program, then we could perhaps adapt the assertion code on the test side to accept all correct orderings?

Reply:
done, could be removed